src/selfplay/loop.cc

```diff
@@ -983,12 +1007,51 @@ void ProcessFile(const std::string& file, SyzygyTablebase* tablebase,
         }
       }
 
-      std::string fileName = file.substr(file.find_last_of("/\\") + 1);
-      TrainingDataWriter writer(outputDir + "/" + fileName);
-      for (auto chunk : fileContents) {
-        // Don't save chunks that just provide move history.
-        if ((chunk.invariance_info & 64) == 0) {
-          writer.WriteChunk(chunk);
+      if (!outputDir.empty()) {
+        std::string fileName = file.substr(file.find_last_of("/\\") + 1);
+        TrainingDataWriter writer(outputDir + "/" + fileName);
+        for (auto chunk : fileContents) {
+          // Don't save chunks that just provide move history.
+          if ((chunk.invariance_info & 64) == 0) {
+            writer.WriteChunk(chunk);
+          }
+        }
+      }
+
+      // Output data in Stockfish plain format.
+      if (!nnue_plain_file.empty()) {
+        static Mutex mutex;
+        std::ostringstream out;
+        if (newInputFormat != -1) {
+          PopulateBoard(
+              static_cast<pblczero::NetworkFormat::InputFormat>(newInputFormat),
+              PlanesFromTrainingData(fileContents[0]), &board, &rule50ply,
+              &gameply);
+        } else {
+          PopulateBoard(input_format, PlanesFromTrainingData(fileContents[0]),
+                        &board, &rule50ply, &gameply);
+        }
+        history.Reset(board, rule50ply, gameply);
+        for (int i = 0; i < fileContents.size(); i++) {
+          auto chunk = fileContents[i];
+          // Format is v6 and position evaluated.
+          if (chunk.visits > 0) {
+            Position p = history.GetPositionAt(i);
+            out << AsNnueString(p, MoveFromNNIndex(chunk.best_idx),
```

Comment:
MoveFromNNIndex isn't 'transform aware' - so if you use canonical input data this can generate illegal moves for the position.